Expensify/App
# Add OptionSelector and NewGroup page

### Details

Adds the NewGroup page. This page will not yet be accessible apart from directly navigating to the page. Once this is merged we will use it as a template to create the "New Chat" and "Search" pages which are essentially pared down versions of the "New Group" page which is the most complex.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/152283

### Tests
1. Navigate to `#/new/group`
1. Verify that there is a "RECENTS" section that displays the last 5 users who you have chatted with and a "CONTACTS" section that displays a list of people you may know or have chatted with. All options should have avatars, display names (if available), and emails. **Note:** None of these should have multiple avatars.
1. Verify all options have an empty circle to the right indicating that multiple can be selected
1. Verify there is a button at the bottom of the panel that says "Create Group"
1. Verify that typing into the input will filter both the list of contacts and recent users
1. Verify that when no contacts are available to select the text changes to a message suggesting you invite a user.
1. Enter a valid email or phone and verify a new option appears
1. Select the option and verify the user is added to the top of the list and they appear as "selected".
1. Select one recent user and one user from the contacts
1. Verify each one is removed from their respective list and moved to the top
1. Tap "Create Group" and verify you are brought to the chat with all three users.
1. Verify we are redirected back to a report view when the `x` in the top right corner is selected

### Tested On


**Only tested the below platforms that this did not affect the builds since 
this page is not accessible apart from direct access via URL**

### Screenshots
![2021-01-25_14-47-36](https://user-images.githubusercontent.com/32969087/105784422-df006480-5f1c-11eb-90b9-1a9ff0f146e0.png)
![2021-01-25_14-50-24](https://user-images.githubusercontent.com/32969087/105784441-e9baf980-5f1c-11eb-807e-21ac53ec6dfb.png)
![2021-01-25_14-50-39](https://user-images.githubusercontent.com/32969087/105784450-ee7fad80-5f1c-11eb-9d54-5e3b502c482a.png)
![2021-01-25_14-50-43](https://user-images.githubusercontent.com/32969087/105784458-f17a9e00-5f1c-11eb-93c4-367da0412a4c.png)
![2021-01-25_15-02-37](https://user-images.githubusercontent.com/32969087/105785193-65697600-5f1e-11eb-995e-187a47d91961.png)
![Screen Shot 2021-01-25 at 3 32 49 PM](https://user-images.githubusercontent.com/32969087/105787218-a06da880-5f22-11eb-9c7d-db677c09dc6c.png)
![Screen Shot 2021-01-25 at 3 32 40 PM](https://user-images.githubusercontent.com/32969087/105787219-a19ed580-5f22-11eb-9bfa-60a33d575c35.png)


**Note:** Yes, I'm aware the selected / unselected bit looks bad it's just a placeholder for the real UI that will come later.
